### ISSUE TITLE
protoToDataFrame maps Enum types to their String name/value

### ISSUE THREAD

**Issue by lokkju** [scalapb/sparksql-scalapb#7]
As the title says; is this expected behavior?

**Comment by thesamet** [comment]
Yes, the SQL syntax does not support any representation of enums, or predefined numeric constants. So I thought that strings would be better than ints.

**Comment by lokkju** [comment]
Hmm.  Ok, I ran into this when trying to roundtrip some protobuf through the spark.write/spark.read; I thought SparkSQL supported custom UDTs, and it seems that custom type is being used when calling spark.CreateDataset in to the DemoSpec; that is why you have the UDFs in there.

This looks promising: https://github.com/dongjinleekr/spark-dataset/blob/master/src/main/scala/com/github/dongjinleekr/spark/dataset/Titanic.scala
